libs/models/src/lib/user.ts

```diff
@@ -0,0 +1,13 @@
+import { Document } from 'mongoose';
+import { CreateUserDto } from './create-user.dto';
+
+export interface User extends CreateUserDto, Document {
+  _id: string;
+  states: UserState[];
```

Comment:
I'm a bit confused about the fact that the state of a user is not an individual object but instead many of them. Why is it that a user has multiple *states*?

I suggest we reconsider this decision and instead make a user have an individual state. This state could hold things like "paymentStatus", "logos", "referredUsers", "credits", etc.

Therefore, I think the data model could be slightly refactored to something like this:

```
User {
  state: UserState;
}

UserState {
  logos: UserLogo;
}

UserLogo {
  imageUrl: string;
  name: string; <-- note that I'm not using 'completed' because a "UserLogo" is understood to be completed
}
```

Reply:
sure, I missed this one. Will make the changes